packages/xforms-engine/src/instance/PrimaryInstance.ts

```diff
@@ -279,4 +297,14 @@ export class PrimaryInstance<
 
 		return Promise.resolve(result);
 	}
+
+	loadMediaResources(resourceURL: string): Promise<MediaResource> {
```

Comment:
This class provides simple functions to access high-level information, like the active language, and could be extended to retrieve media files, such as images or videos. That's why I added the `loadMediaResources` function here. 

To optimize performance, it avoids querying all media files during the initial `LoadForm` phase, as finding media URLs requires traversing a complex node tree. Instead, it leverages the engine’s XML processing phase when it creates an instance, where it already navigates the node tree to prepare data for all question types. By collecting media information during this process, the engine avoids a second, redundant tree traversal.

Later, we can revisit and find a way to retrieve all media URLs at the initial `LoadForm` phase.